feat: adding Register Page

thuanvmse161224 opened the pull request:
Register Page is similar to Login Page, so it was fast developing this

thuanvmse161224 committed:
feat: adding Register Page
Register Page is similar to Login Page, so it was fast developing this

danielNg91 commented on src/pages/Register/RegisterForm/EmailTextField.tsx:
```diff
@@ -0,0 +1,27 @@
+
+
+// eslint-disable-next-line no-useless-escape
+const EMAIL_REGEX = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
```
reuse validation, components from login form

thuanvmse161224 replied:
check new commit [refactor: Extract textfields for reusability]
no error were seen, output the same
commit msg has description about what i did, please review

thuanvmse161224 replied:
If looking good, then i shall re-locate component into component space, so that login and forgot password can use Email text field too. Its logic doesn't differ from each other

dg-panh commented on src/pages/Register/RegisterForm/styles.ts:
I dont understand why need to create theme here. Already have folder to store all theme. Just use it to style text color

thuanvmse161224 replied:
There isn't just text, but border and checkbox. It has other styling css which would be cumbersome to style. Modifying the primary with themeprovider to override default primary would be faster
that's what i think

thuanvmse161224 committed:
refactor: Extract textfields for reusability

1. Extract EmailTextField as a functional component:
   - Using generic type parameter <T> for flexibility

2. Extract PasswordTextField as a functional component:
   - Using generic type parameter <T> for flexibility

3. Update helperText handling in EmailTextField and PasswordTextField:
   - Set helperText to an empty string if error.message is undefined

4. <T extends FieldValues>:
   - register: UseFormRegister requires FieldValues

5. 'email' as Path<T> and 'password' as Path<T>:
   - Explicitly specifies 'email' and 'password' as Path<T> for TS

TLDR: This commit improves reusability, clarifies helperText

thuanvmse161224 committed:
refactor: Reorganize Email and Password TextFields

1. Moved Email and Password TextFields to components folder
2. Updated references in Login Form to use the new components.
3. Removed old Email and Password TextField.

dg-panh reviewed (commented):
LGTM
